execution/aggregate/khashaggregate.go

```diff
@@ -61,10 +61,14 @@ func NewKHashAggregate(
 		compare = func(f float64, s float64) bool {
 			return f < s
 		}
-	} else {
+	} else if aggregation == parser.BOTTOMK {
```

Comment:
Maybe we should error out if some other, unsupported aggregation is passed? Also, it will protect us against any potential bugs because previously we were always having this `compare` function.